src/python/grpcio/grpc/beta/implementations.py

```diff
@@ -105,10 +106,37 @@ def metadata_call_credentials(metadata_plugin, name=None):
     A CallCredentials object for use in a GRPCCallOptions object.
   """
   if name is None:
-    name = metadata_plugin.__name__
```

Comment:
Woops, this one slipped by me. I think your change is the right way to go for non-function callables (which I think I'd not considered at the time). The problem is that we expect most users to just write a new function and pass it in. Functions have `__name__` attributes that are meaningful, but their `__class__` attributes are always `function`. So we'd at least want to maintain support for that use case.